src/Symfony/Component/Console/Application.php

```diff
@@ -116,24 +116,41 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         try {
             $exitCode = $this->doRun($input, $output);
         } catch (\Exception $e) {
-            if (!$this->catchExceptions) {
-                throw $e;
-            }
+            if (null !== $this->dispatcher) {
+                $event = new ConsoleExceptionEvent($this->runningCommand, $input, $output, $e, $e->getCode());
+                $this->dispatcher->dispatch(ConsoleEvents::ERROR, $event);
 
-            if ($output instanceof ConsoleOutputInterface) {
-                $this->renderException($e, $output->getErrorOutput());
-            } else {
-                $this->renderException($e, $output);
+                $e = $event->getException();
```

Comment:
I realize that you want a listener to ERROR to be able to "handle" the exception so that things don't continue. To do this, the listener will call `$event->setException(null)`, which just looks like a weird way to do it.

Perhaps a `$event->setExceptionAsHandled()` or something to signal that the Application shouldn't continue and render the exception? It would also simplify the `$exitCode` code below, since there *will* be an exception still.